CurlWrapper.php

```diff
@@ -296,6 +300,34 @@ public function getTransferInfo($key = null)
     }
 
     /**
+     * Gets the response headers from the last transfer
+     *
+     * If $key is given, returns its value. Otherwise, returns an associative array of the response headers
+     *
+     * @param string $key
+     * @throws CurlWrapperException
+     * @return array|string
+     */
+    public function getResponseHeaders($key = null)
+    {
+        if (empty($this->responseHeaders)) {
+            throw new CurlWrapperException('There are no response headers. Did you do the request?');
+        }
+
+        $this->responseHeaders = $this->parseHeaders($this->responseHeaders);
```

Comment:
Because we reuse `$this->responseHeaders` that function will blow up after second call, which is no good, right? :) And even if the function was a bit different - why parse headers on every call, especially when one of the way of using the function is to call it with `key`?

That function needs to check variable with parsed headers and take data from it if data is available or parse raw headers _once_ and put parsed data to parsed headers variable.

And of course with that scenario we need to reset raw and parsed headers before any new request.